feat(sanitise): Sanitation level

gmertes opened the pull request:
## Description
Add three levels of sanitation:

1. current behaviour (default)
2. remove hostnames but keep the rest of the url, paths and filenames
3. remove everything

 Useful when sharing checkpoints with the public. Even though our URLs are generally protected, there's no reason to keep them in public metadata.

***As a contributor to the Anemoi framework, please ensure that your changes include unit tests, updates to any affected dependencies and documentation, and have been tested in a parallel setting  (i.e., with multiple GPUs). As a reviewer, you are also responsible for verifying these aspects and requesting changes if they are not adequately addressed. For guidelines about those please refer to https://anemoi.readthedocs.io/en/latest/***

By opening this pull request, I affirm that all authors agree to the [Contributor License Agreement.](https://github.com/ecmwf/codex/blob/main/Legal/contributor_license_agreement.md)

floriankrb commented:
I would use a more generic name than `remove_urls`, such as `paranoia_level=2`(default would be 1). This is because I imagine you will add later `remove_filenames` and `remove_uris`, and `remove_database_names`. 

`paranoia_level` may not be the most appropriate name, of course.

gmertes commented:
Added the level argument with the 3 levels + tests. Also cleaned up the docstrings a bit, the private function docstrings were just adding useless clutter.

anaprietonem reviewed (dismissed):
Gert this LGTM, and seems like you addressed Florian's comments so should be good to merge
